File: frontend/demo/component/app-layout/app-layout-navbar.ts
import 'Frontend/demo/init'; // hidden-source-line
import '@vaadin/app-layout';
import '@vaadin/app-layout/vaadin-drawer-toggle';
import '@vaadin/horizontal-layout';
import '@vaadin/icon';
import '@vaadin/icons';
import { css, html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/demo/theme';
import { patchAppLayoutNavigation } from './app-layout-helper'; // hidden-source-line

@customElement('app-layout-navbar')
export class Example extends LitElement {
  static override styles = css`
    h1 {
      font-size: 1.125rem;
      left: 1.5rem;
      margin: 0;
      position: absolute;
    }

    vaadin-tabs {
      margin: auto;
    }
  `;

  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    applyTheme(root);
    return root;
  }

  /* prettier-ignore */ protected firstUpdated() { // hidden-source-line
    patchAppLayoutNavigation(this.shadowRoot!.querySelector('vaadin-horizontal-layout')!); // hidden-source-line
  } // hidden-source-line

  protected override render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-app-layout>
        <h1 slot="navbar">MyApp</h1>
        <vaadin-horizontal-layout
          slot="navbar"
          style="width: 100%; height: 2.25rem; justify-content: center; gap: 0.5rem"
        >
          <a
            href="/dashboard"
            style="display: flex; align-items: center; padding: 0 1rem; font-weight: 500; text-decoration: none"
          >
            Dashboard
          </a>
          <a
            href="/orders"
            style="display: flex; align-items: center; padding: 0 1rem; font-weight: 500; text-decoration: none"
          >
            Orders
          </a>
          <!-- end::snippet[] -->
          <a
            href="/customers"
            style="display: flex; align-items: center; padding: 0 1rem; font-weight: 500; text-decoration: none"
          >
            Customers
          </a>
          <a
            href="/products"
            style="display: flex; align-items: center; padding: 0 1rem; font-weight: 500; text-decoration: none"
          >
            Products
          </a>
          <!-- tag::snippet[] -->
        </vaadin-horizontal-layout>
      </vaadin-app-layout>
      <!-- end::snippet[] -->
    `;
  }
}

File: src/main/java/com/vaadin/demo/component/applayout/AppLayoutNavbar.java
package com.vaadin.demo.component.applayout;

import com.vaadin.demo.DemoExporter; // hidden-source-line
import com.vaadin.flow.component.applayout.AppLayout;
import com.vaadin.flow.component.html.H1;
import com.vaadin.flow.component.orderedlayout.FlexComponent;
import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
import com.vaadin.flow.router.Route;
import com.vaadin.flow.router.RouterLink;

@Route("app-layout-navbar")
// tag::snippet[]
public class AppLayoutNavbar extends AppLayout {

    public AppLayoutNavbar() {
        H1 title = new H1("MyApp");
        title.getStyle().set("position", "absolute").set("left",
                "var(--vaadin-gap-l)");

        HorizontalLayout navigation = getNavigation();
        navigation.getElement()
                .executeJs("window.patchAppLayoutNavigation(this);"); // hidden-source-line

        addToNavbar(title, navigation);
    }
    // end::snippet[]

    private HorizontalLayout getNavigation() {
        HorizontalLayout navigation = new HorizontalLayout();
        navigation.setWidthFull();
        navigation
                .setJustifyContentMode(FlexComponent.JustifyContentMode.CENTER);
        navigation.setHeight("2.25rem");
        navigation.getStyle().set("gap", "0.5rem");
        navigation.add(createLink("Dashboard"), createLink("Orders"),
                createLink("Customers"), createLink("Products"));
        return navigation;
    }

    private RouterLink createLink(String viewName) {
        RouterLink link = new RouterLink();
        link.add(viewName);
        // Demo has no routes
        // link.setRoute(viewClass.java);
        link.getStyle().set("display", "flex").set("align-items", "center")
                .set("padding", "0 1rem").set("font-weight", "500")
                .set("text-decoration", "none");
        // hidden-source-line: workaround to make text color work
        link.getElement().setAttribute("href", viewName); // hidden-source-line

        return link;
    }

    public static class Exporter extends DemoExporter<AppLayoutNavbar> { // hidden-source-line
    } // hidden-source-line
      // tag::snippet[]
}
// end::snippet[]
